SharedPackages/VPN/Sources/VPN/Diagnostics/ConnectionTesting.swift

```diff
@@ -26,6 +26,9 @@ import Foundation
 @MainActor
 protocol ConnectionTesting: AnyObject {
 
+    /// Handler called with connection test results.
+    var resultHandler: (@MainActor (NetworkProtectionConnectionTester.Result) -> Void)? { get set }
```

Comment:
To make the tester injectable, we'll initialize this after the PTP's `super.init()`, but we'll also need to assign this callback after `super.init()` rather than on initialization.